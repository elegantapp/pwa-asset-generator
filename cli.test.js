const execa = require('execa');
const file = require('./helpers/file');

test('throws error when input is not provided', async () => {
  try {
    await execa.sync('./cli.js', []);
  } catch (e) {
    expect(e.stderr).toContain('Please specify a URL or file path as a source');
  }
});

test('creates an output folder when output path does not exist', async () => {
  jest.setTimeout(30000);
  const tempFolderName = 'temp';
  await execa.sync('./cli.js', [
    './static/logo.png',
    tempFolderName,
    '-s=false',
    '--icon-only',
  ]);
  expect(await file.pathExists(tempFolderName)).toBe(true);
});

test('creates images', async () => {
  jest.setTimeout(30000);
  let flags = `
    --background="rgba(255, 255, 255, .5)"
    --scrape=false
    --icon-only
    `;

  flags = flags
    .trim()
    .replace(/(?<==)"|(?<!\\)"$/gm, '')
    .replace(/\\"/g, '"')
    .split('\n');

  const { stdout } = await execa('./cli.js', [
    './static/logo.png',
    './static',
    ...flags,
  ]);

  expect(stdout).toContain(
    'Below is the icons content for your manifest.json file. You can copy/paste it manually',
  );
  expect(stdout).toContain(
    'Below is the splash screen content for your index.html file. You can copy/paste it manually',
  );
});
