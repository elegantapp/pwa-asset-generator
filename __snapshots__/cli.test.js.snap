// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates icons and splash screens when both only flags exist 1`] = `
"
[
  {
    \\"src\\": \\"temp/manifest-icon-192.png\\",
    \\"sizes\\": \\"192x192\\",
    \\"type\\": \\"image/png\\"
  },
  {
    \\"src\\": \\"temp/manifest-icon-512.png\\",
    \\"sizes\\": \\"512x512\\",
    \\"type\\": \\"image/png\\"
  }
]


<link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"temp/apple-icon-180.png\\">
<link rel=\\"apple-touch-icon\\" sizes=\\"167x167\\" href=\\"temp/apple-icon-167.png\\">
<link rel=\\"apple-touch-icon\\" sizes=\\"152x152\\" href=\\"temp/apple-icon-152.png\\">
<link rel=\\"apple-touch-icon\\" sizes=\\"120x120\\" href=\\"temp/apple-icon-120.png\\">

<meta name=\\"apple-mobile-web-app-capable\\" content=\\"yes\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2048-2732.png\\"
    media=\\"(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2732-2048.png\\"
    media=\\"(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1668-2388.png\\"
    media=\\"(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2388-1668.png\\"
    media=\\"(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1668-2224.png\\"
    media=\\"(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2224-1668.png\\"
    media=\\"(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1536-2048.png\\"
    media=\\"(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2048-1536.png\\"
    media=\\"(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1242-2688.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2688-1242.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1125-2436.png\\"
    media=\\"(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2436-1125.png\\"
    media=\\"(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-828-1792.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1792-828.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1242-2208.png\\"
    media=\\"(device-width: 414px) and (device-height: 736px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2208-1242.png\\"
    media=\\"(device-width: 414px) and (device-height: 736px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-750-1334.png\\"
    media=\\"(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1334-750.png\\"
    media=\\"(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-640-1136.png\\"
    media=\\"(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1136-640.png\\"
    media=\\"(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
"
`;

exports[`generates icons only 1`] = `
"
[
  {
    \\"src\\": \\"temp/manifest-icon-192.png\\",
    \\"sizes\\": \\"192x192\\",
    \\"type\\": \\"image/png\\"
  },
  {
    \\"src\\": \\"temp/manifest-icon-512.png\\",
    \\"sizes\\": \\"512x512\\",
    \\"type\\": \\"image/png\\"
  }
]


<link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"temp/apple-icon-180.png\\">
<link rel=\\"apple-touch-icon\\" sizes=\\"167x167\\" href=\\"temp/apple-icon-167.png\\">
<link rel=\\"apple-touch-icon\\" sizes=\\"152x152\\" href=\\"temp/apple-icon-152.png\\">
<link rel=\\"apple-touch-icon\\" sizes=\\"120x120\\" href=\\"temp/apple-icon-120.png\\">

<meta name=\\"apple-mobile-web-app-capable\\" content=\\"yes\\">
"
`;

exports[`generates landscape splash screens only 1`] = `
"

<meta name=\\"apple-mobile-web-app-capable\\" content=\\"yes\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2732-2048.png\\"
    media=\\"(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2388-1668.png\\"
    media=\\"(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2224-1668.png\\"
    media=\\"(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2048-1536.png\\"
    media=\\"(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2688-1242.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2436-1125.png\\"
    media=\\"(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1792-828.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2208-1242.png\\"
    media=\\"(device-width: 414px) and (device-height: 736px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1334-750.png\\"
    media=\\"(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1136-640.png\\"
    media=\\"(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
"
`;

exports[`generates portrait splash screens only 1`] = `
"

<meta name=\\"apple-mobile-web-app-capable\\" content=\\"yes\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2048-2732.png\\"
    media=\\"(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1668-2388.png\\"
    media=\\"(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1668-2224.png\\"
    media=\\"(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1536-2048.png\\"
    media=\\"(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1242-2688.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1125-2436.png\\"
    media=\\"(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-828-1792.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1242-2208.png\\"
    media=\\"(device-width: 414px) and (device-height: 736px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-750-1334.png\\"
    media=\\"(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-640-1136.png\\"
    media=\\"(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
"
`;

exports[`generates splash screens only 1`] = `
"

<meta name=\\"apple-mobile-web-app-capable\\" content=\\"yes\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2048-2732.png\\"
    media=\\"(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2732-2048.png\\"
    media=\\"(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1668-2388.png\\"
    media=\\"(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2388-1668.png\\"
    media=\\"(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1668-2224.png\\"
    media=\\"(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2224-1668.png\\"
    media=\\"(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1536-2048.png\\"
    media=\\"(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2048-1536.png\\"
    media=\\"(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1242-2688.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2688-1242.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1125-2436.png\\"
    media=\\"(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2436-1125.png\\"
    media=\\"(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-828-1792.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1792-828.png\\"
    media=\\"(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1242-2208.png\\"
    media=\\"(device-width: 414px) and (device-height: 736px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-2208-1242.png\\"
    media=\\"(device-width: 414px) and (device-height: 736px) and (-webkit-device-pixel-ratio: 3) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-750-1334.png\\"
    media=\\"(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1334-750.png\\"
    media=\\"(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-640-1136.png\\"
    media=\\"(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2) and (orientation: portrait)\\">
<link rel=\\"apple-touch-startup-image\\"
    href=\\"temp/apple-splash-1136-640.png\\"
    media=\\"(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2) and (orientation: landscape)\\">
"
`;
